certbot/certbot/compat/misc.py

```diff
@@ -122,8 +123,9 @@ def execute_command(cmd_name: str, shell_cmd: str, env: Optional[dict] = None) -
     :param str shell_cmd: shell command to execute
     :param dict env: environ to pass into Popen
 
-    :returns: `tuple` (`str` stderr, `str` stdout)
+    :returns: `tuple` (`int` returncode, `str` stderr, `str` stdout)
     """
+    # TODO: redesign execute_command and execute_command_status in Certbot 2.0 API.
```

Comment:
Tbh, what we have here isn't too messy, despite having one unused function (that is, we don't necessarily need to redesign, just deprecate/streamline). If we are going to clean it up in 2.0 though, we should make an associated issue and link to it.

Reply:
Totally. There's quite a few hooks-related issues to consider for 2.0. Not sure how we're tracking that, other than the 2.0 milestone.

How do we do this deprecation thing? Just `warning.warning(..., DeprecationWarning)` right in `execute_command`, and create an issue to remove in 2.0 milestone?

Reply:
Yes, that's all correct, just it's `warnings.warn(..., PendingDeprecationWarning)`.